gridGeneration.py

```diff
@@ -0,0 +1,259 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--maxDepth', type = str, default='', help='max depth of a k-d tree')
+    parser.add_argument('--countNum', type = str, default='', help='a count value')
+    parser.add_argument('--gridPercent', type = str, default='', help='a grid percentage')
+    parser.add_argument('--maxCount', type = str, default='', help='maximum count to the second k-d tree')
+    parser.add_argument('--outFolder', type = str, default='', help='path to an ouput folder')
+
+    args = parser.parse_args()
+    maximum_level = args.maxDepth
+    folder_path = args.outFolder
+    count_num = int(args.countNum)
+    grid_percent = float(args.gridPercent)
+
+    print(os.path.join(folder_path, 'histogram'))
+    
+    path = 'histogram'
+    geojson_path = 'geojson'
+    
+    if not os.path.exists(os.path.join(folder_path, path)):
+        print('Create the histogram directory !!')
+        os.makedirs(os.path.join(folder_path, path))
+    if not os.path.exists(os.path.join(folder_path, geojson_path)):
+        print('Create the geojson directory !!')
+        os.makedirs(os.path.join(folder_path, geojson_path))
+    
+    max_count = 0
+    flag_val = False
+    input_data = None
+    optimal_count_list = None
+    optimal_grid_size_list = None
+    first_depth = 0
+    big_initial_area = 0.0
+
+    if args.maxCount:
+        max_count = int(args.maxCount)
+
+    # read L-2 dissimilarity from stand input
+    for line in sys.stdin.readlines():
+        input_data = ast.literal_eval(line)
+    
+    # save the 2d list as a file
+    with open(os.path.join(folder_path, os.path.basename(input_data['inFolder']) + '.csv' )  , "w") as out_f:
+        writer = csv.writer(out_f)
+        writer.writerows(input_data['nameNum'])
+    
+    # perform the 1st k-d tree
+    for depth_count in range(1, int(maximum_level) + 1):
+        # build k-d tree
+        tree_cons = kdTree(depth_count, input_data['iniBb'], input_data['data'], 1)
+        out_tree = tree_cons.tree_building()
+
+        # get leaves given a K-D tree
+        bb_collec = tree_cons.get_leaves(out_tree)
+
+        # get counts
+        hist, gridid_collec= tree_cons.counts_calculation()
+        # Cell size computation
+        cell_size_computation(depth_count, bb_collec)
+        del tree_cons
+
+        util = Utility(hist)
+        filename = os.path.join(os.path.join(folder_path, path), 'level-' + str(depth_count) + '.png')
+        # probability distribution
+        out_distribution, count_list, count_zero_list, cell_num = util.distribution_computation(filename)
+        # write out a Geojson file
+        util.geojson_write(depth_count, bb_collec, os.path.join(folder_path, geojson_path), cell_num, input_data['iniArea'], None, 'tree_v1', flag_val)
+        del util
+
+        # stop condition (the over 90% (parameter) of cells is less than 10 (parameter) (the count value))
+        if len(count_zero_list) != 0:
```

Comment:
All instances where you check to see if the length of a list is 0 or not can be replaced with either:

`if not list` for empty or `if list` if not empty.

Reply:
modified the syntax to "if list".